tools/agent_tool/src/cpp/main.cpp

```diff
@@ -0,0 +1,245 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file agent_tool.cpp
+ *
+ */
+
+#include <algorithm>
+#include <thread>
+
+#include <cpp_utils/event/SignalEventHandler.hpp>
+#include <cpp_utils/Log.hpp>
+#include <ddsrouter_core/types/address/Address.hpp>
+
+#include <amlip_cpp/node/wan/ClientNode.hpp>
+#include <amlip_cpp/node/wan/ServerNode.hpp>
+#include <amlip_cpp/node/wan/TurnNode.hpp>
+
+#include "user_interface/arguments_configuration.h"
+
+enum EntityType
+{
+    CLIENT,
+    SERVER,
+    REPEATER
+};
+
+int main(
+        int argc,
+        char** argv)
+{
+    eprosima::utils::event::SignalEventHandler<eprosima::utils::event::Signal::sigint> sigint_handler;
+
+    // Help message formatting settings
+    int columns;
+
+#if defined(_WIN32)
+    char* buf = nullptr;
+    size_t sz = 0;
+    if (_dupenv_s(&buf, &sz, "COLUMNS") == 0 && buf != nullptr)
+    {
+        columns = strtol(buf, nullptr, 10);
+        free(buf);
+    }
+    else
+    {
+        columns = 80;
+    }
+#else
+    columns = getenv("COLUMNS") ? atoi(getenv("COLUMNS")) : 80;
+#endif // if defined(_WIN32)
+
+    // Parameter definition
+    EntityType entity_type = EntityType::CLIENT;
+    std::string connection_address = "127.0.0.1";
+    std::string listening_address = "127.0.0.1";
+    std::string ip;
+    int domain = 0;
+    int connection_port = 12121;
+    int listening_port = 12121;
+    int port;
+    char* name = "agent_tool";
```

Comment:
use a string or `const char`:

`warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]   74 |     char* name = "agent_tool";`